skill-tree/src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java

```diff
@@ -290,4 +290,96 @@ public void testAPIReturn404_OnEndorsementNotFound() {
+    public void testAPIReturn200_OnEndorsementSearchByUserID() {
+        String userIDString = "f13ac7a0-76ab-4215-8bfc-2dd5d9f8ebeb";
+
+            given()
+                .when()
+                    .get("/v1/endorsements?userID=" + userIDString)
+                .then()
+                    .statusCode(200)
+                    .contentType("application/json")
+                    .body("content", everyItem(hasKey("user_id")))
+                    .body("content.user_id", everyItem(equalTo(userIDString)));
+    }
+
+    @Test
+    @DisplayName("Return 200 with the endorsements of a particular skill given skillID")
+    public void testAPIReturn200_OnEndorsementSearchBySkillID() {
+        String skillIDString = "7a6b8876-44e3-4b18-8579-79e9d4a5f0c9";
+            given()
+                .when()
+                    .get("/v1/endorsements?skillID=" + skillIDString)
+                .then()
+                    .statusCode(200)
+                    .contentType("application/json")
+                    .body("content", everyItem(hasKey("skill_id")))
+                    .body("content.skill_id", everyItem(equalTo(skillIDString)));
+    }
+
+    @Test
+    @DisplayName("Return 200 with the endorsements matching the given userID and skillID")
+    public void testAPIReturn200_OnEndorsementSearchByBothUserIDAndSkillID() {
+        String userIDString = "73e0b7c4-d128-4e53-9501-0e7f4ff5a261";
+        String skillIDString = "7a6b8876-44e3-4b18-8579-79e9d4a5f0c9";
+
+            given()
+                .when()
+                    .get("/v1/endorsements?skillID=" + skillIDString + "&userID=" + userIDString)
+                .then()
+                    .statusCode(200)
+                    .contentType("application/json")
+                    .body("content", everyItem(hasKey("skill_id")))
+                    .body("content.skill_id", everyItem(equalTo(skillIDString)))
+                    .body("content", everyItem(hasKey("user_id")))
+                    .body("content.user_id", everyItem(equalTo(userIDString)));
+    }
+
+    @Test
+    @DisplayName("Return 204 when there are no endorsements present for the given userID")
+    public void testAPIReturn204_OnEndorsementSearchWithValidUserID() {
+        String userIDString = UUID.randomUUID().toString();
+
+            given()
+                .when()
+                    .get("/v1/endorsements?userID=" + userIDString)
+                .then()
+                    .statusCode(204);
+    }
+
+    @Test
+    @DisplayName("Return 204 when there are no endorsements present for the given skillID")
+    public void testAPIReturn204_OnEndorsementSearchWithValidSkillID() {
+        String skillIDString = UUID.randomUUID().toString();
+
+            given()
+                .when()
+                    .get("/v1/endorsements?skillID=" + skillIDString)
+                .then()
+                    .statusCode(204);
+    }
+
+    @Test
+    @DisplayName("Return 400 given an invalid userID")
```

Comment:
in which context user_id is invalid, this seems not correct to me

Reply:
This is testing the scenario where user_id given is not a UUID, for example /v1/endorsements?userID=abc. I will change the test description to make it more clear.

Reply:
I have changed the description, can you please review it once?